test/scripts/e2e_subs/e2e-teal.sh

```diff
@@ -148,6 +150,18 @@ ${gcmd} clerk compile ${TEMPDIR}/true3.teal -o ${TEMPDIR}/true3.lsig
 cp ${TEMPDIR}/true3.lsig ${TEMPDIR}/true2.lsig
 printf '\x02' | dd of=${TEMPDIR}/true2.lsig bs=1 seek=0 count=1 conv=notrunc
 
+# Try to compile with source map, and check that map is correct.
+# Since the source map contains info about the file path,
+# we need to do this in place and clean up the file later.
+${gcmd} clerk compile ${TEAL}/quine.teal -m
+if ! diff ${TEAL}/quine.map ${TEAL}/quine.teal.map > /dev/null; then
+    rm ${TEAL}/quine.teal.*
```

Comment:
@algochoi Since the diff output is sent to `/dev/null` and the generated source map is deleted, how do you expect a maintainer to debug a failing test?

Unless I've missed a debugging facility, I see a few ways to make life easier for maintainers:
* Retain `diff` output.  From my local testing, `diff` produces output _only_ when a diff exists (which fails the test).
* Either in addition to or in place of the above option:  Write the generated source map file to a `generated` (or similarly named) dir that's added to `.gitignore`.  The proposal removes the need to delete the generated output.  The maintainer can inspect generated output to debug a failing test.

If given an option, it seems to me retaining `diff` output is the simpler change.

Aside:  Assuming the PR continues deleting the generated source map, I optionally propose refactoring to delete unconditionally via signal trapping.
* https://stackoverflow.com/questions/687014/removing-created-temp-files-in-unexpected-bash-exit points to a couple of examples and related links.
* Additionally, there's an example in `e2e_subs`:  https://github.com/algorand/go-algorand/blob/07b2fbb25906003e496e9f20f50c3d28ff35fa08/test/scripts/e2e.sh#L53-L64